### PR TITLE
semgrep 0.43.0

### PR DESCRIPTION
**Pull request by brendongo**
[`action-homebrew-bump-formula`](https://github.com/dawidd6/action-homebrew-bump-formula)

---

Created with `brew bump-formula-pr`.

~~`resource` blocks may require updates.~~

### CODE REVIEW

**Comment by carlocab** [comment]
Looks like the formula needs updating:
```
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - _build/default/cli/Main.exe
```
The error comes from this line:

https://github.com/Homebrew/homebrew-core/blob/82d8455fd632836855ecad76b40135188ce9f91c/Formula/semgrep.rb#L156

I also have two questions.
1. Do the resources not need updating?
2. Can we use 4.10.2 here? I think that might enable building an ARM bottle. Can we just use the `ocaml` compiler from the `ocaml` formula?
https://github.com/Homebrew/homebrew-core/blob/82d8455fd632836855ecad76b40135188ce9f91c/Formula/semgrep.rb#L130

**Comment by brendongo** [comment]
@carlocab whoops looks like you're right. Updated the formula and dependencies. We haven't verified everything works with 4.10.2 but we're evaluating 4.12.0 in the near future.

**Comment by carlocab** [comment]
>Want me to include changing that in this PR?

Yes please. I'll flag the ARM node to attempt a build.

File: Formula/semgrep.rb [hunk 1]
```diff
@@ -127,7 +127,7 @@ def install
       ENV["PREFIX"] = prefix
 
       system "opam", "init", "--no-setup", "--disable-sandboxing"
-      ENV.deparallelize { system "opam", "switch", "create", "ocaml-base-compiler.4.10.0" }
+      ENV.deparallelize { system "opam", "switch", "create", "ocaml-base-compiler.4.10.2" }
```
Comment (SMillerDev) [inline Formula/semgrep.rb:130]: Would this work?
```suggestion
      ENV.deparallelize { system "opam", "switch", "create", "ocaml-base-compiler.#{Formula["ocaml"].version}" }
```

Comment (brendongo) [reply Formula/semgrep.rb:130]: Is that guaranteed to be 4.10.2?

**Comment by brendongo** [comment]
> > Want me to include changing that in this PR?
> 
> Yes please. I'll flag the ARM node to attempt a build.

I think it successfully built on ARM!

File: Formula/semgrep.rb (hunk 1 above)
Comment (carlocab) [reply Formula/semgrep.rb:130]: It isn't; it's likely to be 4.12 relatively soon.

Comment (brendongo) [reply Formula/semgrep.rb:130]: Probably best to leave as 4.10.2 literal for now then?

**Review by carlocab — APPROVED** [review]
Thanks, @brendongo.

**Pull request closed without merging**